### PR TITLE
Fix compiler warnings for the iOS platform.

### PR DESCRIPTION
**Pull request by DeepakArora76**
Fix compiler warnings for iOS platform. The related issue to it is #2715.

### CODE REVIEW

**Comment by kencu** [comment]
this won't work like this, I believe.

The macros are defined, but defined to "0", on the other platforms. 

So TARGET_OS_IPHONE is defined, but defined to "0", building on Mac.

You will need to adjust your tests.

This has crept into the code in other places as well and will need fixing elsewhere too.

**Comment by saghul** [comment]
Do you know when `TARGET_OS_IPHONE` started to be ddefined to 0? I think this hasn't been the case in the past, has it?

**Comment by kencu** [comment]
Since at least 10.6:
{{{
cd /usr/include

$ ag TARGET_OS_IPHONE .
TargetConditionals.h
63:        TARGET_OS_IPHONE        - Generate code will run under iPhone OS which 
92:    #define TARGET_OS_IPHONE            0 
}}}

**Comment by saghul** [comment]
Ah right. `if TARGET_OS_IPHONE` is ok, which I think we use in some places. TThis one iss incorrecct indeed.

**Comment by kencu** [comment]
there's at least one more in the close_nocancel function to be fixed. Haven't searched for others as yet. Thanks!

**Comment by bnoordhuis** [comment]
@DeepakArora76 Can you include the feedback from the discussion above and rebase + tidy up the commit logs?

File: src/unix/fs.c [hunk 1]
```diff
@@ -151,7 +151,7 @@ extern char *mkdtemp(char *template); /* See issue #740 on AIX < 7 */
     }                                                                         \
     else {                                                                    \
       uv__fs_work(&req->work_req);                                            \
-      return req->result;                                                     \
+      return (int)req->result;                                                \
```
Comment (vtjnash) [inline src/unix/fs.c:154]:
```suggestion
      return req->result;                                                     \
```
This warning is generally correct: we are causing the wrong answer to be returned here. (cf #1501)

**Comment by vtjnash** [comment]
Code deleted by 47effc4bd39e4e13c59102bab3e789783725ecb6

**Pull request closed without merging**